File: python/5_kyu/where_my_anagrams_at.py
# Where My Anagrams At?
# https://www.codewars.com/kata/523a86aa4230ebb5420001e1/train/python


# Solution 1: Sorted and Equal
def anagrams(word, words):
    return [i for i in words if sorted(i) == sorted(word)]

print(anagrams("abba", ['aabb', 'abcd', 'bbaa', 'dada']))
print(anagrams("racer", ['crazer', 'carer', 'racar', 'caers', 'racer']))
print(anagrams("laser", ['lazing', 'lazy',  'lacer']))

File: javascript/5_kyu/where_my_anagrams_at.js
// Where My Anagrams At?
// https://www.codewars.com/kata/523a86aa4230ebb5420001e1/train/javascript

// Solution 1: Sort and Compare

// Create a sorting function to sort string
function sort_string(x) {
    return x.split("").sort().join("");
}
// run the solution using the created sorting function
function anagrams(word, words) {
    var ans = [];
    for (var i = 0; i < words.length; i++) {
        if (sort_string(words[i]) === sort_string(word)) {
            ans.push(words[i]);
        }
    }

    return ans;
}

console.log(anagrams("abba", ['aabb', 'abcd', 'bbaa', 'dada']));
console.log(anagrams("racer", ['crazer', 'carer', 'racar', 'caers', 'racer']));
console.log(anagrams("laser", ['lazing', 'lazy',  'lacer']));